horovod/common/operations.cc

```diff
@@ -231,7 +242,9 @@ void PerformOperation(Response response) {
     if (!status.ok()) {
       for (auto& e : entries) {
         timeline.End(e.tensor_name, nullptr);
-        e.callback(status);
+        if (!state.joined) {
```

Comment:
In `cuda_operations.cc` you just check that the callback is not null.  Maybe we can do the same here to make it more general?